test/cluster/mzcompose.py

```diff
@@ -2546,3 +2547,68 @@ def worker(c: Composition, i: int) -> None:
         assert (
             p99 < p99_limit
         ), f"p99 is {p99:.2f}s, should be less than {p99_limit:.2f}s"
+
+
+def workflow_test_incident_70(c: Composition) -> None:
+    """
+    Test incident-70.
+    """
+    num_conns = 10
+    persist_reader_lease_duration_in_sec = 30
+    data_scale_factor = 10
+
+    c.down(destroy_volumes=True)
+    c.up("materialized")
+
+    c.sql(
+        f"ALTER SYSTEM SET max_connections = {num_conns + 5};",
+        port=6877,
+        user="mz_system",
+    )
+
+    c.sql(
+        f"ALTER SYSTEM SET persist_reader_lease_duration = '{persist_reader_lease_duration_in_sec}s';",
+        port=6877,
+        user="mz_system",
+    )
+
+    c.sql(
+        f"""
+        CREATE SOURCE gen FROM LOAD GENERATOR TPCH (SCALE FACTOR {data_scale_factor}, TICK INTERVAL '100ms') FOR ALL TABLES;
+
+        CREATE MATERIALIZED VIEW mv_lineitem_count AS SELECT count(*) FROM lineitem;
```

Comment:
This is close to what I tested against, but:

- I did not use a tick interval... the load generator was just used to create a static dataset.
- The scale factor was either 100 or perhaps a bit lower (80?).
- I spun up multiple materialized views with this shape, to ramp up the load. (Up to a dozen.)

Reply:
Thanks! I will try to reproduce it with that.

However, I am a bit sceptical whether a scale factor of this size will be viable and sustainable in CI. I haven't found an existing test with a scale factor in this range and with the current configuration, a scale factor of 10 and a single mz already OOMs. (CC @philip-stoev)

Reply:
Yeah, agree that this might be untenable. It's possible that you could compensate for a smaller dataset side by even more dataflows downstream, but that may give you memory pressure from another direction...